src/intervals/arithmetic.jl

```diff
@@ -394,9 +406,10 @@ function mid(a::Interval{T}) where T
     a.lo == -∞ && return nextfloat(a.lo)
     a.hi == +∞ && return prevfloat(a.hi)
 
-    # @assert 0 ≤ α ≤ 1
-
-    return 0.5 * (a.lo + a.hi)
+    midpoint = 0.5 * (a.lo + a.hi)
+    isfinite(midpoint) && return midpoint
+    # Fallback in case of overflow: a.hi + a.lo == +∞ or a.hi + a.lo == -∞
+    return 0.5 * a.lo + 0.5 * a.hi
```

Comment:
Doesn't this line work for all cases? Is this for efficiency?

Reply:
Yes it works for all cases, it is simply a bit less efficient (on floating point operation more). I simply tried to preserve the previous implementation.

However it looks like the added structure add a prohibitive cost so I should only keep the last line

```julia
julia> function f(a::Interval)
           midpoint = 0.5 * (a.lo + a.hi)
           isfinite(midpoint) && return midpoint
       end
f (generic function with 1 method)

julia> g(a::Interval) = 0.5 * a.lo + 0.5 * a.hi
g (generic function with 1 method)

julia> @btime f.(rr) ;
  248.527 μs (20032 allocations: 391.97 KiB)

julia> @btime g.(rr) ;
  20.160 μs (26 allocations: 79.23 KiB)
```

(should have run this tests before committing)

Reply:
**Erratum:** Actually some tests (in `minimal_mid_tests` in the `ITF1788_tests\libieeep1788_tests_num.jl` test suite) fail with the `0.5 * a.lo + 0.5 * a.hi` line. The reason for that is the fact that
```julia
0.5 * nextfloat(0.0) == 0.5 * prevfloat(0) == 0
```

Therefore (with the longer version) we would have
```julia
julia> X = (2*prevfloat(0.0))..nextfloat(0.0)
[-1.4822e-323, 9.88132e-324]

julia> mid(X)
-5.0e-324
```

but the value `0.0` is expected (this is phrased in term of hexadecimal number in the test suite but I find that version more readable). So I think I did the right choice (I should definitively document it in the code though).

Reply:
OK great, thanks.